### PR TITLE
[PATCH v3] travis: add ubuntu 18.04 compilation test

### PR DESCRIPTION
**Pull request by muvarov**
Signed-off-by: Maxim Uvarov <maxim.uvarov@linaro.org>

### CODE REVIEW

**Comment by lumag** [comment]
I'd suggest to move them out of 'build only' stage to proper matrix entries. Build only should contain  5 entries, as we have only 5 concurrent builders. Adding more entries to that stage might delay 'test' stage.

**Comment by muvarov** [comment]
v2: moved from build only to test.

**Comment by Bill-Fischofer-Linaro** [comment]
PR has checkpatch errors, otherwise looks good.
```
0001-travis-add-ubuntu-18.04-compilation-test.patch
+perl ./scripts/checkpatch.pl 0001-travis-add-ubuntu-18.04-compilation-test.patch
No codespell typos will be found - file '/usr/share/codespell/dictionary.txt': No such file or directory
No structs that should be const will be found - file '/home/travis/build/Linaro/odp/scripts/const_structs.checkpatch': No such file or directory
ERROR: trailing whitespace
#19: FILE: .travis.yml:190:
+                - stage: test $
total: 1 errors, 0 warnings, 0 checks, 15 lines checked
NOTE: For some of the reported defects, checkpatch may be able to
      mechanically convert to the typical style using --fix or --fix-inplace.
NOTE: Whitespace errors detected.
      You may wish to use scripts/cleanpatch or scripts/cleanfile
0001-travis-add-ubuntu-18.04-compilation-test.patch has style problems, please review.
NOTE: Ignored message types: AVOID_EXTERNS BIT_MACRO BLOCK_COMMENT_STYLE COMPARISON_TO_NULL CONSTANT_COMPARISON CONST_STRUCT DEPRECATED_VARIABLE NEW_TYPEDEFS PREFER_KERNEL_TYPES PREFER_PRINTF PREFER_SCANF SPLIT_STRING SSCANF_TO_KSTRTO UNNECESSARY_PARENTHESES VOLATILE
NOTE: If any of the errors are false positives, please report
      them to the maintainer, see CHECKPATCH in MAINTAINERS.
```

File: .travis.yml [hunk 1]
```diff
@@ -187,6 +187,15 @@ jobs:
...
                               ${DOCKER_NAMESPACE}/travis-odp-lng-ubuntu_16.04 /odp/scripts/ci/build_${ARCH}.sh
+                - stage: test 
+                  env: ARCH=x86_64 CC=clang UBUNTU_VERS="18.04"
+                  install:
+                          - true
+                  script:
+                          - if [ -z "${DOCKER_NAMESPACE}" ] ; then export DOCKER_NAMESPACE="opendataplane"; fi
+                          - docker run  -i -t -v `pwd`:/odp
+                              -e CC="${CC}"
+                              ${DOCKER_NAMESPACE}/travis-odp-lng-ubuntu_${UBUNTU_VERS} /odp/scripts/ci/build_${ARCH}.sh
```
Comment (lumag) [inline .travis.yml]: What about just putting it into plain environment matrix? With UBUNTU_VERSION being environment entry?

Comment (muvarov) [reply .travis.yml]: hm, let me do a try...

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**